openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -180,3 +206,57 @@ def _email_url_parser(url_name, extra_param=None):
     dashboard_url_path = reverse(url_name) + extra_param if extra_param else reverse(url_name)
     dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
     return urlparse.urlunparse(dashboard_link_parts)
+
+
+def get_credit_providers_by_course(course_key):
+    """Get the course information from ecommerce and parse the data to get providers.
+
+    Arguments:
+        course_key (CourseKey): The identifier for the course.
+
+    Returns:
+        list, containing course providers.
+    """
+    credit_config = CreditConfig.current()
+
+    providers_data = None
+    cache_key = None
+    if credit_config.is_cache_enabled:
+        cache_key = '{key_prefix}.{course_key}'.format(
+            key_prefix=credit_config.CACHE_KEY, course_key=unicode(course_key)
+        )
+        providers_data = cache.get(cache_key)
+
+    if providers_data is not None:
+        return providers_data
+
+    providers_data = []
+
+    try:
+        user = User.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+        response = ecommerce_api_client(user).courses(unicode(course_key)).get(include_products=1)
+    except Exception:  # pylint: disable=broad-except
+        log.exception("Failed to receive data from the ecommerce course API for Course ID '%s'.", unicode(course_key))
+        return providers_data
+
+    if not response:
+        log.info("No Course information found from ecommerce API for Course ID '%s'.", unicode(course_key))
+        return providers_data
+
+    for product in response.get('products'):
+        providers_data += [
+            attr.get('value') for attr in product.get('attribute_values') if attr.get('name') == 'credit_provider'
+        ]
+
+    # Hit the CreditProvider method to get the all active providers with complete information.
+    providers_information = CreditProvider.get_credit_providers()
+    providers_display_names = []
+    for provider_id in providers_data:
+        providers_display_names += [
+            provider['display_name'] for provider in providers_information if provider['id'] == provider_id
+        ]
```

Comment:
You can do this more cleanly. For example:
```python
providers = CreditProvider.get_credit_providers()
for provider in providers:
    if provider['id'] in provider_ids:
        provider_display_names += provider['display_name']
```
If `providers_data` contains provider IDs, rename it `provider_ids` for readability. Same with `providers_information`, which contains `providers`.